### PR TITLE
Add video streaming capabilities API

### PR DESCRIPTION
**Pull request by jhludwig**
This proposal provides a video streaming capabilites API to SDL apps. This is achieved by HMI notification of video capabilities to SDL core.

### CODE REVIEW

**Comment by jhludwig** [comment]
@theresalech this is ready for review

**Comment by theresalech** [comment]
As decided during the Steering Committee meeting on May 2, Xevo will revise this proposal to be aligned with SDL #166 , so that this proposal can be brought into review during the Steering Committee meeting on May 9 (2017-05-09).

**Comment by jhludwig** [comment]
@theresalech per request, we have rewritten this to be consistent with SDL #166, it is now ready for review

File: proposals/nnnn-video-streaming-capabilities.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+This proposal should not have impacts on existing code since it only adds some capability parameters in the structures. Of course, HMIs are encouraged to implement capturing these capabilities of the head units and filling in the structures with appropriate values.
+
+## Alternatives considered
+
+The struct could be added to `RegisterAppInterface` response instead. This is not considered as a good approach since the response now includes large capability information and it does not seem to scale well.
+
+## Notes
+
+While this proposal includes an idea of `preferredResolution`, it does not provide a mean to determine the actual resolution of the video stream. In other word, a HMI cannot know whether a SDL app uses a resolution of `preferredResolution` or not without inspecting the H.264 video stream (such as SPS NAL unit). The HMI needs video stream's resolution information in order to send out touch events with correct coordinate on the video image. This issue is outside the scope of this proposal, and may be addressed by another one.
```
Comment (theresalech) [inline proposals/nnnn-video-streaming-capabilities.md]: Please change "mean" to "means" and "other word" to "other words".

**Comment by jhludwig** [comment]
@theresalech changes made per your request

**Pull request merged**